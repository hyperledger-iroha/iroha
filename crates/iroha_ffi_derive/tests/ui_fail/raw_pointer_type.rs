use iroha_ffi::FfiType;

/// Raw pointer type
#[derive(Clone, Copy, PartialEq, Eq, FfiType)]
#[repr(C)]
pub struct FfiStruct1(*mut u32);

/// Raw pointer type
#[derive(Clone, Copy, PartialEq, Eq, FfiType)]
pub enum FfiEnum1 {
    A,
    B(*mut u32),
}

fn main() {}
