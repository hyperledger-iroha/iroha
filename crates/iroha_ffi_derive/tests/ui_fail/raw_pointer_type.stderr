error: Raw pointer found. If the pointer doesn't own the data, attach `#[ffi_type(unsafe {non_owning})` to the field. Otherwise, mark the entire type as opaque with `#[ffi_type(opaque)]`
 --> tests/ui_fail/raw_pointer_type.rs:6:23
  |
6 | pub struct FfiStruct1(*mut u32);
  |                       ^^^^^^^^

error: Raw pointer found. If the pointer doesn't own the data, attach `#[ffi_type(unsafe {non_owning})` to the field. Otherwise, mark the entire type as opaque with `#[ffi_type(opaque)]`
  --> tests/ui_fail/raw_pointer_type.rs:12:7
   |
12 |     B(*mut u32),
   |       ^^^^^^^^
